### PR TITLE
Do not grant all privs to the admin - changes game behavior

### LINKED ISSUE #6456: Discussion: Host server without being admin
I'm sure this has come up before. I did a quick search through the issues with "host server", but nothing related came up...

Currently it is not possible to use the UI to setup a quick game between two local players without one being necessarily an admin - and is annoying since some behaviors changes (digging does not increase the count in the inventory, etc).

In order to do that you'd have to run a separate server, with a separate config file, just to change the admin name. So what good is host server in the UI in the first place?

I think this is:
- Hard to grasp for new users
- Somewhat annoying if you just want a quick game

Not sure what easy options there are, but at least we should be able to have a separate config to declare the server admin from a player that happens to play locally on the same machine.

Maybe have a "name" and "adminname" config...?

This is just for discussion. This must have come up before somewhere.

### PR DESCRIPTION
**Pull request by lhofhansl**
Do not grant all privs to the admin. Admin gets all privs to manage a server and the "privs" right to grant more privs.

**Was:**
Absolutely simplest way to fix #6456.
One can optionally name a specific user to be administrator using the admin_name property.
If set, that user will be administrator, if not set (or empty) the default behavior is retained (the name property indicates the administrator)

### CODE REVIEW

**Comment by lhofhansl** [comment]
I've been using this for my one-off games, and it works fine.

**Comment by lhofhansl** [comment]
Could even default the admin's name to "admin" and advise server owners to change it.

**Comment by nerzhul** [comment]
if you default it to admin your emptiness test is a noop

**Comment by SmallJoker** [comment]
Better not setting "admin" as default, it would only cause trouble for people who would either want to use this default name or those who could abuse it. Keeping it empty is the best choice.

File: builtin/game/chatcommands.lua [hunk 1]
```diff
@@ -82,7 +82,8 @@ core.register_chatcommand("me", {
 core.register_chatcommand("admin", {
 	description = "Show the name of the server owner",
 	func = function(name)
-		local admin = minetest.settings:get("name")
+		local admin =  minetest.settings:get("admin_name") ~= "" and minetest.settings:get("admin_name")
```
Comment (sfan5) [inline builtin/game/chatcommands.lua]: extra space

Comment (SmallJoker) [reply builtin/game/chatcommands.lua]: \* core.settings
For these two lines

Comment (lhofhansl) [reply builtin/game/chatcommands.lua]: I like the shortening...

**Comment by lhofhansl** [comment]
So is this good?
One consideration: Anything specific to do for the singleplayer? Should that use always be admin regardless of the admin_name setting? (but I wouldn't want to hard code "singleplayer" anywhere)

File: builtin/game/chatcommands.lua (hunk 1 above)
Comment (lhofhansl) [reply builtin/game/chatcommands.lua]: Alas, the shortening would return true if amin_name is not empty (not the value)

**Comment by lhofhansl** [comment]
Anything else I should do here?

**Comment by jastevenson303** [comment]
You can start a minetestserver without name, just not in the GUI.

For example:
```
mkdir test && cd test
touch test.conf
minetestserver --worldname test --config test.conf
```
Then open client and connect localhost and issue /admin:
screenshot_20170928_154215

Is this PR about making this possible GUI-side?

**Comment by lhofhansl** [comment]
I understand that :)

**Comment by jastevenson303** [comment]
My question is whether you're trying to implement an already existent feature from the CLI-side to the GUI-side, or implementing something entirely new and different with "admin_name" setting in particular.

**Comment by lhofhansl** [comment]
Oh I see.

It's just to make the existing features more usable.
- In the case of the GUI you cannot start a local games without being the admin. That's annoying.
- For the command line server, you cannot start a server and join the server from the same Minetest install - although that's probably less common.

In the end it's just that the name property is used for different purposes, on the client it indicates the player on the server it indicates the admin.

**Comment by Megaf** [comment]
:+1: for the Idea. not sure about the implementation.

**Comment by rubenwardy** [comment]
I sense that this may cause problems. It would be better to fix the use of privilege to make them not modify gameplay, but to give a permission

**Comment by jastevenson303** [comment]
In singleplayer mode, most privileges are missing from the `singleplayer` user.  However `grant` is there, so they can `/grantme server` if they like.  Could this same method be implemented in a hosted multiplayer game?

Definitely an important issue, and I'm sure many have run into it and not reported it, myself included.

**Comment by lhofhansl** [comment]
Another way is to just grant the permission to grant any permission to any user to the admin. That way the admin does not have any special permissions until he/she grant them to him/herself.

**Comment by lhofhansl** [comment]
yeah, I think that's best. Just grant the admin the "privs" priv default, and be done. Admin can than grant self (or anybody else) any privs needed.

Comments?

**Comment by lhofhansl** [comment]
Hmmm... The problem is the "creative" priv, which is not actually a priv (it's not in privileges.lua), but somehow the admin has it, but singleplayer does not. Looking...

**Comment by lhofhansl** [comment]
Ahh. That's in minetest_game, and it's not given to singleplayer.

**Comment by lhofhansl** [comment]
How about this? Needs matching change in minetest_game

**Comment by lhofhansl** [comment]
But I think it's still bad or at least confusing the same config "name" is used in two different ways.

**Comment by lhofhansl** [comment]
It turns out there're other mods that register a priv called "creative" :(
Should we whilelist admin perms instead?

**Comment by lhofhansl** [comment]
Sorry for the spam... Or I can just re-use give_to_singleplayer, but that's not quite right.

**Comment by rubenwardy** [comment]
> Should we whilelist admin perms instead?

It's better as it is

**Comment by jastevenson303** [comment]
@lhofhansl I typically just edit my last message if no one has posted below me.  : )

I'll be happy to test this and your other PRs tomorrow, or in the next coming days.  Good stuff!

**Comment by lhofhansl** [comment]
Perhaps we default to give_to_singleplayer but let it override with give_to_admin?
I'd like to move this forward, the current behavior is really quite annoying.

(I don't think it's a feature, but a bugfix)

**Comment by lhofhansl** [comment]
Did just that. Games' #1905 (or fixing unified inventory, etc) is no longer needed.

**Pull request merged**